src/metabase/mbql/normalize.cljc

```diff
@@ -639,25 +639,30 @@
 
 (defn- canonicalize-mbql-clauses
   "Walk an `mbql-query` an canonicalize non-top-level clauses like `:fk->`."
-  [mbql-query]
-  (walk/prewalk
-   (fn [x]
-     (cond
-       (map? x)
-       (m/map-vals canonicalize-mbql-clauses x)
-
-       (not (mbql-clause? x))
-       x
-
-       :else
-       (try
-         (canonicalize-mbql-clause x)
-         (catch #?(:clj Throwable :cljs js/Error) e
-           (log/error (i18n/tru "Invalid clause:") x)
-           (throw (ex-info (i18n/tru "Invalid MBQL clause: {0}" (ex-message e))
-                           {:clause x}
-                           e))))))
-   mbql-query))
+  [x]
+  (cond
+    (map? x)
+    (m/map-vals canonicalize-mbql-clauses x)
+
+    (not (mbql-clause? x))
+    (if (sequential? x)
+      (into (empty x) (map canonicalize-mbql-clauses) x)
```

Comment:
This reverses lists and lazy sequences, I have to change this, if those can appear here.